lib/puppet/defaults.rb

```diff
@@ -43,6 +48,11 @@ def self.default_diffargs
           directory, but if it's running as any other user, it defaults to being
           in the user's home directory.",
     },
+    :codedir  => {
+        :default  => "$puppetdir/code",
+        :type     => :directory,
+        :desc     => "TODO ncw",
```

Comment:
@nfagerlund Not sure how we want to describe these directories now.  Suggestions welcome :smiley: